rueidisotel/trace.go

```diff
@@ -53,10 +66,26 @@ func TraceAttrs(attrs ...attribute.KeyValue) Option {
 	}
 }
 
+// WithMeterProvider sets the MeterProvider for the otelclient.
+func WithMeterProvider(provider metric.MeterProvider) Option {
+	return func(o *otelclient) {
+		o.meterProvider = provider
+	}
+}
+
+// WithTracerProvider sets the TracerProvider for the otelclient.
+func WithTracerProvider(provider trace.TracerProvider) Option {
+	return func(o *otelclient) {
+		o.tracerProvider = provider
+	}
+}
+
 type otelclient struct {
-	client rueidis.Client
-	mAttrs []attribute.KeyValue
-	tAttrs []attribute.KeyValue
+	client         rueidis.Client
+	mAttrs         []attribute.KeyValue
+	tAttrs         []attribute.KeyValue
+	meterProvider  metric.MeterProvider
+	tracerProvider trace.TracerProvider
```

Comment:
Besides the providers, we should also move `tracer`, `cscMiss` and `cscHits` into this struct.

Reply:
ah... got it. I trying to add test codes and i faced the problem what you said